src/zcl_abapgit_tadir.clas.abap

```diff
@@ -51,9 +51,20 @@ CLASS zcl_abapgit_tadir IMPLEMENTATION.
           ls_srcsystem    LIKE LINE OF lt_srcsystem,
           ls_exclude      LIKE LINE OF lt_excludes.
 
+    DATA: last_package TYPE devclass VALUE cl_abap_char_utilities=>horizontal_tab.
+    DATA: lo_folder_logic TYPE REF TO zcl_abapgit_folder_logic.
+
     FIELD-SYMBOLS: <ls_tdevc> LIKE LINE OF lt_tdevc,
                    <ls_tadir> LIKE LINE OF rt_tadir.
 
+    "Determine Packages to Read
+    DATA: lt_packages TYPE zif_abapgit_sap_package=>ty_devclass_tt.
+    IF iv_ignore_subpackages = abap_true.
+      INSERT iv_package INTO TABLE lt_packages.
+    ELSE.
+      lt_packages = zcl_abapgit_factory=>get_sap_package( iv_package )->list_subpackages( ).
+      INSERT iv_package INTO lt_packages INDEX 1.
+    ENDIF.
```

Comment:
this part can be rewritten to have 2 less statements?

```
IF iv_ignore_subpackages = abap_false.
  lt_packages = zcl_abapgit_factory=>get_sap_package( iv_package )->list_subpackages( ).
ENDIF.
INSERT iv_package INTO lt_packages INDEX 1.
```

Reply:
Yes it can. I just found it easier to understand with the if/else construct and it doesn't make a difference performance wise.
But I can compress it. No problem.